metalarchives_importer.user.js

```diff
@@ -0,0 +1,242 @@
+
+    // Tracks
+    /*discs = [
+            {
+                title,
+                format,
+                tracks = [
+                    { number, title, duration, artist_credit },
+                    ...
+                ]*/
+    var releaseNumber=1;
+    release.discs = new Array();
+    release.discs.push(new Object());
+    release.discs[releaseNumber-1].tracks=new Array();
+    release.discs[releaseNumber-1].format=ReleaseFormat[rdata["Format"]];
+    var tracksline= $('table.table_lyrics tr.even,table.table_lyrics tr.odd');
+    var trackslinelength=tracksline.length;
+    console.dir(tracksline);
+    tracksline.each(function(index,element){
+        var trackNumber= $.trim(element.children[0].textContent).replace('.',"");
+        if( trackNumber == "1" && trackNumber != index+1 ){
+            releaseNumber++;
+            release.discs.push(new Object());
+            release.discs[releaseNumber-1].tracks = new Array();
+            release.discs[releaseNumber-1].format=ReleaseFormat[rdata["Format"]];
+        }
+        console.log(element.children[1]);
+        var track = {
+            'number':trackNumber,
+            'title': element.children[1].textContent,
+            'duration': element.children[2].textContent,
+            'artist_credit': [release.artist_credit]
+        };
+        release.discs[releaseNumber-1].tracks.push(track);
+    });
+    console.log(release);
+    return release;   
+}
+
+// Insert links in page
+function insertLink(release) {
+
+    var edit_note = 'Imported from ' + window.location.href;
+    var parameters = MBImport.buildFormParameters(release, edit_note);
+    var innerHTML = MBImport.buildFormHTML(parameters);
+
+    $('h2.band_name').append(innerHTML);
+    $("form[action='//musicbrainz.org/release/add']").css("padding","initial");
+
+}
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//                                   Metal Archives -> MusicBrainz mapping                                                   //
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+/* 
+release.type 	primary type release 	secondary release type 
+on MA				on MB 	
+
+Full-length 		Album				Compilation		                  
+Live album			Single				Demo
+Demo				EP					DJ-mix
+Single				Broadcast	Interview
+EP					Other		Live
+Video							Audiobook
+Boxed set							Mixtape/Street
+Split								Remix
+Video/VHS (legacy)					Soundtrack
+Compilation							Spokenword
+Split video
+*/
+//ReleaseTypes[MAtype]=["primary type","secondary type on mb"];
+var ReleaseTypes = new Array();
+ReleaseTypes["Full-length"]=["Album"];
+ReleaseTypes["Live album"]=["Album", "Live"];
+ReleaseTypes["Demo"]=["Album","Demo"];
+ReleaseTypes["Single"]=["Single"];
+ReleaseTypes["EP"]=["EP"];
+ReleaseTypes["Compilation"]=["Album", "Compilation"];
+ReleaseTypes["Split"]=["Album"];
```

Comment:
"Collaboration" is missing, and prevents import of e.g. http://www.metal-archives.com/albums/Metallica/Lulu/320007